api/lib/validate.js

```diff
@@ -166,8 +178,95 @@ function validateOrderbook(orderbook) {
   }
 }
 
+function validateSettings(settings) {
+  if (typeof settings !== 'object') {
+    throw error('Invalid parameter: settings');
```

Comment:
It's cleaner, fewer lines of code in the checking side and basically what exceptions are designed for.